profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/features/ServerApi.java

```diff
@@ -0,0 +1,212 @@
+   
+   @Named("server:volume:delete")
+   @DELETE
+   @Path("/{serverId}/volumes/{volumeId}")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   void detachVolume(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId, @PathParam("volumeId") String volumeId);
+   
+   @Named("server:volume:get")
+   @GET
+   @Path("/{serverId}/volumes/{volumeId}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @ResponseParser(VolumeApi.VolumeParser.class)
+   Volume getVolume(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId, @PathParam("volumeId") String volumeId);
+   
+   @Named("server:cdrom:list")
+   @GET
+   @Path("/{serverId}/cdroms")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   @SelectJson("items")
+   List<Image> listAttachedCdroms(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId);
+   
+   @Named("server:cdrom:attach")
+   @POST
+   @MapBinder(AttachCdromRequestBinder.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @ResponseParser(ImageApi.ImageParser.class)
+   Image attachCdrom(@PayloadParam("cdrom") Server.Request.AttachCdromPayload payload);
+   
+   @Named("server:cdrom:delete")
+   @DELETE
+   @Path("/{serverId}/cdroms/{cdRomId}")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   void detachCdrom(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId, @PathParam("cdRomId") String cdRomId);
+   
+   @Named("server:cdrom:get")
+   @GET
+   @Path("/{serverId}/cdroms/{cdRomId}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @ResponseParser(ImageApi.ImageParser.class)
+   Image getCdrom(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId, @PathParam("cdRomId") String cdRomId);
+   
+   @Named("server:reboot")
+   @POST
+   @Path("/{serverId}/reboot")
+   void rebootServer(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId);
+   
+   @Named("server:start")
+   @POST
+   @Path("/{serverId}/start")
+   void startServer(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId);
+   
+   @Named("server:stop")
+   @POST
+   @Path("/{serverId}/stop")
+   void stopServer(@PathParam("dataCenterId") String dataCenterId, @PathParam("serverId") String serverId);
+   
+   static final class ServerParser extends ParseJson<Server> {
+      @Inject ServerParser(Json json) {
+         super(json, TypeLiteral.get(Server.class));
+      }
+
+      @Override
+      public <V> V apply(InputStream stream, Type type) throws IOException {
+         String data = Strings2.toStringAndClose(stream);
+         
+         Gson gson = new Gson();
+         Type mapType = new TypeToken<Map<String, Object>>(){}.getType();
+         Map<String, String> jsonMap = gson.fromJson(data, mapType);
+         
+         Pattern p = Pattern.compile(".*/datacenters/(.*)/servers.*");
```

Comment:
Maybe move this as a constant above? So that the pattern is compiled only once, and not on every method call.